File: microservices/python/command_handlers/create_container.py
'''
Created on Jan 26, 2018

@author: pawel
'''

import json

from systembus.eventbus import EventBus

# system settings
events_cfg = json.load(open('../../systembus/events.json'))


def createContainerCommandHandler(command):
    print('handling command', command)
    # TODO do stuff here...

    # get event names to emit
    command_name = command['command_name']
    event_names = [event_name for event_name, cfg in events_cfg.items() \
                                    if command_name in cfg['on_behalf_of']]

    with EventBus(events_config = events_cfg) as eb: 
        for event_name in event_names:
            event = {
                'event_name' : event_name,
            }
            eb.send(event)
            print('event emited', event)

File: microservices/node/command_handlers/create_container.js
var commands_cfg = require("../../../systembus/commands.json");
    events_cfg = require("../../../systembus/events.json");
    handlers_cfg = require("../../../systembus/handlers.json");

    emit = require('../event_producer.js').emit;


function createContainerCommandHandler(command) {
    console.log('handling command', JSON.stringify(command));
    // TODO do stuff here...

    var command_name = command['command_name']
        handler_name = commands_cfg[command_name]['handler']

    handlers_cfg[handler_name]['event_names'].forEach(function(event_name){
        event = {
            'event_name' : event_name,
            'uuid' : command['uuid'],
            'source' : command['source'],
            }
        emit(event_name, event)
    });
}

module.exports = {createContainerCommandHandler}